lighthouse-core/test/audits/resource-summary-test.js

```diff
@@ -0,0 +1,93 @@
+/**
+ * @license Copyright 2017 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const Audit = require('../../audits/resource-summary.js');
+const assert = require('assert');
+const trace = require('../fixtures/traces/progressive-app-m60.json');
+const networkRecordsToDevtoolsLog = require('../network-records-to-devtools-log.js');
+const URL = require('url').URL;
+
+/* eslint-env jest */
+
+function generateRequest(url, type, size, baseUrl = 'http://google.com/') {
```

Comment:
I'm guessing you got this one from `total-byte-weight`? :)

as author of that test, I've come to deeply regret using positional arguments and apologize for the bad example I've set for the world 😳 (Also with the introduction of `network-records-to-devtools-log.js` more recently much of it is just unnecessary)


for your use case we can *probably* get away with very simple network records as just objects

```js
const devtoolsLogs = networkRecordsToDevtoolsLog([
  {url: 'http://example.com/file.html', resourceType: 'Document', transferSize: 30},
  {url: 'http://example.com/app.js', resourceType: 'Script', transferSize: 10},
  {url: 'http://third-party.com/script.js', resourceType: 'Script', transferSize: 50},
  {url: 'http://third-party.com/file.jpg', resourceType: 'Image', transferSize: 70},
])